linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/JSGen.scala

```diff
@@ -104,14 +106,27 @@ private[emitter] final class JSGen(val semantics: Semantics,
   }
 
   def genIsInstanceOf(expr: Tree, typeRef: TypeRef)(
+      implicit globalKnowledge: GlobalKnowledge, pos: Position): Tree = {
+    val useInstanceOf = typeRef match {
+      case ClassRef(className) =>
+        className != NumberClass && // the only non-Object super*class* of hijacked classes
```

Comment:
Could we put this in `MyDefinitions` as a `Set` as well, just so that all the hijacked hierarchy stuff is in one place?